### PR TITLE
 BackendModel extra deletion #1436

### PR DESCRIPTION
**Pull request by vytsci**

### CODE REVIEW

**Comment by vytenizs** [comment]
Please check code with caution. Its late and im tired :D but i need a fix fast.

**Comment by WouterSioen** [comment]
@carakas can you check this? You've done stuff with these extras lately.

**Comment by carakas** [comment]
@WouterSioen I will

File: src/Backend/Core/Engine/Model.php [hunk 1]
```diff
@@ -192,11 +192,13 @@ public static function deleteExtra($module = null, $type = null, array $data = n
         foreach ($extras as $extra) {
             $deleteExtra = true;
 
-            // match by parameters
-            if ($data !== null && $extra['data'] !== null) {
-                $extraData = (array) unserialize($extra['data']);
+            // get extra data
+            $extraData = $extra['data'] !== null?(array)unserialize($extra['data']):null;
```
Comment (jeroendesloovere) [inline src/Backend/Core/Engine/Model.php]: Spaces missing "before" and "after".
"?" => " ? "
":" => " : "

Comment (jeroendesloovere) [inline src/Backend/Core/Engine/Model.php]: Hi @vytsci, can you add missing spaces in your changed line? Thanks

File: src/Backend/Core/Engine/Model.php [hunk 2]
```diff
@@ -192,11 +192,13 @@ public static function deleteExtra($module = null, $type = null, array $data = n
         foreach ($extras as $extra) {
             $deleteExtra = true;
 
-            // match by parameters
-            if ($data !== null && $extra['data'] !== null) {
-                $extraData = (array) unserialize($extra['data']);
+            // get extra data
+            $extraData = $extra['data'] !== null ? (array)unserialize($extra['data']) : null;
```
Comment (vytsci) [inline src/Backend/Core/Engine/Model.php]: @jeroendesloovere is it ok now?

Comment (jeroendesloovere) [inline src/Backend/Core/Engine/Model.php]: Hi,
can you change `(array)unserialize` to `(array) unserialize`.
Thanks

**Comment by WouterSioen** [comment]
thanks

**Pull request merged**